cql3/expr/to_restriction.cc

```diff
@@ -342,8 +342,11 @@ ::shared_ptr<restrictions::restriction> to_restriction(const expression& e,
         throw exceptions::invalid_request_exception(format("Unsupported \"!=\" relation: {}", prepared_binop));
     }
 
-    if (prepared_binop.op == oper_t::IS_NOT && !is<null>(prepared_binop.rhs)) {
-        throw exceptions::invalid_request_exception(format("Unsupported \"IS NOT\" relation: {}", prepared_binop));
+    if (prepared_binop.op == oper_t::IS_NOT) {
+        bool rhs_is_null = is<constant>(prepared_binop.rhs) && as<constant>(prepared_binop.rhs).is_null();
+        if (!rhs_is_null) {
+            throw exceptions::invalid_request_exception(format("Unsupported \"IS NOT\" relation: {}", prepared_binop));
+        }
```

Comment:
I started work to convert IS NOT NULL to be a unary_expression (so we can remove expr::null), but hit snags. Maybe this work can help it along.